Gems/Terrain/Code/Source/Components/TerrainWorldComponent.cpp

```diff
@@ -128,4 +132,50 @@ namespace Terrain
         }
         return false;
     }
-}
+
+    float TerrainWorldConfig::NumberOfSamples(AZ::Vector3* min, AZ::Vector3* max, AZ::Vector2* heightQuery)
+    {
+        float numberOfSamples = ((max->GetX() - min->GetX()) / heightQuery->GetX()) * ((max->GetY() - min->GetY()) / heightQuery->GetY());
+        return numberOfSamples;
+    }
+
+    AZ::Outcome<void, AZStd::string> TerrainWorldConfig::DetermineMessage(float numSamples)
+    {
+        if (numSamples < 4.0f * 1024.0f * 1024.0f)
+        {
+            return AZ::Success();
+        }
+
+        if (numSamples < 8.0f * 1024.0f * 1024.0f)
+        {
+            QMessageBox::information(
+                QApplication::activeWindow(), "Performance Warning",
+                "The number of samples may mean there is a delay in calculating the values", QMessageBox::Ok);
+            return AZ::Success();
+        }
```

Comment:
Discussed offline - if we remove the warning here, we can just get rid of all the Qt dependencies and leave the rest of the changes here alone.  The warning is only a short-term thing anyways, as this will completely change once we address scalability and performance in the terrain system.